packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManager.java

```diff
@@ -141,10 +172,27 @@ public void setBackgroundColor(@NonNull T view, int backgroundColor) {
   @Override
   @ReactProp(name = ViewProps.TRANSFORM)
   public void setTransform(@NonNull T view, @Nullable ReadableArray matrix) {
+    view.setTag(R.id.transform, matrix);
     if (matrix == null) {
       resetTransformProperty(view);
     } else {
-      setTransformProperty(view, matrix);
+      ReadableArray transformOrigin = (ReadableArray) view.getTag(R.id.transform_origin);
+      setTransformProperty(view, matrix, transformOrigin);
+    }
+  }
+
+  @Override
+  @ReactProp(name = ViewProps.TRANSFORM_ORIGIN)
+  public void setTransformOrigin(@NonNull T view, @Nullable ReadableArray transformOrigin) {
+    view.setTag(R.id.transform_origin, transformOrigin);
+    ReadableArray transformMatrix = (ReadableArray) view.getTag(R.id.transform);
+    if (transformMatrix != null) {
+      setTransformProperty(view, transformMatrix, transformOrigin);
+    }
+    if (transformOrigin != null) {
+      view.addOnLayoutChangeListener(this);
+    } else {
+      view.removeOnLayoutChangeListener(this);
```

Comment:
Not sure what the best solution is, but because setters are called in series, we will be calculating the transform twice, once after `setTransform` and again after `setTransformOrigin`.

Reply:
yes, i had this concern. I think we can call in `onAfterUpdateTransaction` when both values are set. While testing I noticed that `onAfterUpdateTransaction` is not called for animating values. So I guess we still call it in setters but use `onAfterUpdateTransaction` for the initial mount.

Reply:
@javache pushed an optimization. This works with animated values as well as fabric. `onAfterUpdateTransaction` is triggered reliably. I did something wrong in my earlier observations. I don't like this [id approach](https://github.com/facebook/react-native/pull/38558/commits/070787f31c9c113eae925d038878aeb5ea4d3247#diff-e0019246f0db288bcbb348337367cc6fc73f1f82bd75c9c8c001762501a7fbe9R176) but can't use instance variables as View manager can manage multiple views. Another approach would be to move the transform stuff in RootViewGroup, but that will require type casting the generic view in Manager and will need to move some files. Let me know wyt!